pkg/webhook/statefulset/validating/statefulset_validation.go

```diff
@@ -44,15 +43,16 @@ func validatePodManagementPolicy(spec *appsv1beta1.StatefulSetSpec, fldPath *fie
 
 func validateReserveOrdinals(spec *appsv1beta1.StatefulSetSpec, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
-
-	if spec.ReserveOrdinals != nil {
-		orders := sets.NewInt(spec.ReserveOrdinals...)
-		if orders.Len() != len(spec.ReserveOrdinals) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.ReserveOrdinals, "reserveOrdinals contains duplicated items"))
-		}
-		for _, i := range spec.ReserveOrdinals {
-			if i < 0 {
-				allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.ReserveOrdinals, fmt.Sprintf("reserveOrdinals contains %d which must be order >= 0", i)))
+	if len(spec.ReserveOrdinals) > 0 {
+		matcher := regexp.MustCompile(`^\d+-\d+$`)
+		for i, elem := range spec.ReserveOrdinals {
+			if elem.Type == intstr.String && !matcher.MatchString(elem.StrVal) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Root(), spec.ReserveOrdinals,
```

Comment:
we should also checking the start and ending string is a valid number, avoid the case like "03-01"